lib/dry/configurable/setting.rb

```diff
@@ -69,6 +69,11 @@ def initialize(name, input: Undefined, default: Undefined, **options)
         evaluate if input_defined?
       end
 
+      def merge(setting)
+        input.merge(setting.input) if input_defined?
+        evaluate
+      end
```

Comment:
Depending on how we adjust `Settings#merge_setting` (per my previous comment), I expect we might want to make some changes here too.

For example, it probably makes sense to move this into `Setting::Nested`, because merging a non-nested setting doesn't really make sense.

At the same time, we should probably raise an `ArgumentError` if the given `setting` param is not also an instance of `Setting::Nested`.

And once we're inside `Setting::Nested`, we could probably make the implementation a lot more direct. The `input_defined?` check won't be needed because `Nested` settings will _always_ have an input. And then because we're no longer having to worry about constructors or default values (not relevant for nested settings), we could probably skip the indirection of `evaluate` and just set the `@value` directly.

You'll have to give it a try to see how these changes shake out, but I think it'd be worth it!